plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c

```diff
@@ -326,8 +321,58 @@ fu_synaptics_rmi_v7_device_write_partition(FuSynapticsRmiDevice *self,
 							     fu_chunk_get_data_sz(chk),
 							     error))
 			return FALSE;
-		fu_progress_step_done(progress);
 	}
+
+	/* write signature */
+	if (g_ptr_array_index(bytes, 1)) {
```

Comment:
this is odd; do you mean `bytes->len > 1`? It's also weird to call it bytes if it's an array.